frontend/src/core/comments/actions.js

```diff
@@ -0,0 +1,91 @@
+/* @flow */
+
+import isEmpty from 'lodash.isempty';
+import NProgress from 'nprogress';
+
+import api from 'core/api';
+import * as history from 'modules/history';
+import * as notification from 'core/notification';
+
+import type { TeamComment } from 'core/api';
+
+
+export const RECEIVE: 'comments/RECEIVE' = 'comments/RECEIVE';
+export const REQUEST: 'comments/REQUEST' = 'comments/REQUEST';
+
+
+export type ReceiveAction = {|
+    +type: typeof RECEIVE,
+    +comments: Array<TeamComment>,
+|};
+export function receive(
+    comments: Array<TeamComment>,
+): ReceiveAction {
+    return {
+        type: RECEIVE,
+        comments,
+    };
+}
+
+
+export type RequestAction = {|
+    +type: typeof REQUEST,
+    +entity: number,
+|};
+export function request(
+    entity: number,
+): RequestAction {
+    return {
+        type: REQUEST,
+        entity,
+    };
+}
+
+
+export function get(entity: number, locale: string): Function {
+    return async dispatch => {
+        dispatch(request(entity));
+
+        // Abort all previously running requests.
+        await api.entity.abort();
+
+        let content = await api.entity.getTeamComments(entity, locale);
+
+        // The default return value of aborted requests is {},
+        // which is incompatible with reducer
```

Comment:
If the request was aborted, it's probably better to not call the `receive` action, isn't it? Since the content will be bogus anyway?

Reply:
That makes sense.  Since this is included in the `get` function of `otherlocales/actions.js` I thought it was a needed check.

I should still call `api.entity.abort()` though, correct?
(history, otherlocales, and machinery all make this call but entities does not)

Reply:
Yes. The reason why we have this abort mechanism is that, if a user changes entities quickly, several requests are sent for helpers tabs content, and because network is unreliable, sometimes the results for request `t` would arrive after results for request `t+1`, thus showing the wrong content. To avoid that, when we start a new request for data, we abort the previously running request.

Note that if you copy/pasted this from somewhere else, it's likely that somewhere else is incorrect as well… :see_no_evil: